app/src/main/kotlin/io/homeassistant/companion/android/notifications/MessagingManager.kt

```diff
@@ -1885,11 +1885,9 @@ class MessagingManager @Inject constructor(
             val action = data[INTENT_ACTION]
             val className = data[INTENT_CLASS_NAME]
             val intentUri = if (!data[INTENT_URI].isNullOrEmpty()) data[INTENT_URI]?.toUri() else null
-            val intent = if (intentUri != null) Intent(action, intentUri) else Intent(action)
+            val intent = Intent(action)
             val type = data[INTENT_TYPE]
-            if (!type.isNullOrEmpty()) {
-                intent.type = type
-            }
+            intent.setDataAndType(intentUri, type)
```

Comment:
@dshokouhi I'm not sure how to test this. 
Before this change if a URI was set and also a type the URI is cleared in favor of the type
<img width="465" height="156" alt="image" src="https://github.com/user-attachments/assets/99336758-a2ce-4077-9597-82f65631b3ae" />

now it is going to keep both.

Reply:
I think to test this you can test the SMS intent to compose a message which accepts a MIME type, IIRC we need to validate the type data is actually present or the receiving app may reject the intent due to invalid data.

https://developer.android.com/guide/components/intents-common#Messaging

a lot of these commands were originally built based off the examples from this page and also the Google maps intent page

https://developer.android.com/guide/components/google-maps-intents#launch-turn-by-turn-navigation

Reply:
So the lint rule was helpful and help us fixed an issue where if the URI and TYPE were set it would simply keep only the TYPE.